### DISCUSSION TITLE
auth().currentUser state after app update

### DISCUSSION THREAD

**Discussion by lulmichal** [invertase/react-native-firebase#5418] [Q&A]
Good Morning!

I have a question about what happens after app version update. We received a complaint from one of the users. Here's the situation. She had an account created with `auth().createUserWithEmailAndPassword` back in 2020. Now we released version update and she updated the app. Now I don't know the exact steps taken yet but she managed to sign in with Facebook credentials. Facebook account is registered with different email than one that was used to sign in in 2020. So naturally the new account was created but here's where the problem begin. Somehow those two accounts merged and has impact on each other. I don't want to dive in into the code because issue might be there but first I need to establish why this could happened (and I'm having a hard time doing that). So my question is this: assuming that user was logged in when she updated the app is it possible that after update `auth().currentUser` wasn't `null` and was actually user data even though we didn't invoke any sign in method yet after the update?

I'm still new to programming so if there are things that I need to add or clarify, please tell.

Best regards.

**Comment by mikehardy** [comment]
> I don't want to dive in into the code

Why not? I don't understand why you wouldn't want to investigate your own code as a strong preference to requesting advice from random people on the internet without providing them any information about your project. That doesn't seem like an efficient path to understanding

There is some reading to do: https://firebase.google.com/docs/auth/web/account-linking

**Reply by lulmichal** [reply]
I meant I don't want to dive in into the code here because I don't know how many lines I'd need to deliver. Also I added that if any additional information is required then I'll be glad to provide it. In my opinion there is no logical flaws in the code so I need to understand how exactly "it" works before making any changes. I think that makes sense especially if I asked specific question.

This reading has been done. I'm not sure I'm following though. As I also mentioned I'm still newbie so explanation would be appreciated. Let's say that I have enabled signing in with email/password and Facebook. I noticed that if first sign in with email/password and then again with Facebook (same email that I used to sign in with email/password) then Firebase will automatically link those two. Is this what is happening (but manually) in the text that you past?

Then again what would be a good practise to implement this into the code? Let's say on my onboarding screen I have "Sign in with Facebook" button that will execute `auth().signInWithCredential(facebookCredentials)`. In case I explained I'm assuming that there already is `auth().currentUser` instance so if it isn't `null` I should implement `linkWithRedirect` for example?

I know it's probably not a good practise but as a quick workaround - if I call `auth().signOut()` at the beginning of my inflate function it would prevent from such an event, right?

Thank you and have a nice day.

**Reply by mikehardy** [reply]
Your second comment reminded me of something I read long ago:

Firebase has the concept of "providers that verify email addresses" and "providers that do not verify email addresses".

If you start with a provider that verifies an email address, then sign in with another provider that is considered by the firebase group to verify email addresses, then firebase will consider both accounts to belong to the same user since the emails are verified, so it will automatically link them

I forget where I read it but I believe it was stackoverflow and it might turn up in a search there - where the firebase group actually respond from time to time to user support queries on generic firebase behavior, which is what this is. react-native-firebase is just wrapping firebase auth from the underlying firebase-android-sdk / firebase-ios-sdk native projects, so if you need more specifics you should inquire there for authoritative information
